### ISSUE TITLE
doc query

### ISSUE THREAD

**Issue by retorquere** [qualified-cactus/promised-db#3]
Just to be sure, in the readme it shows

```
// key range can be created from Index definition
const nameIndex = todoTaskObjectStore.index(TodoTask.NameIndex)
// get all tasks with name greater than "bazz"
const tasksList = nameIndex.getAll(TodoTask.NameIndex.lowerBound("bazz"))

// iterate over keys of index / objectstore
nameIndex.iterator.iterateKeys(async (cursor)=>{
  console.log(cursor.key)  // access index key
  console.log(cursor.primaryKey) // access primary key
},{
  query: TodoTask.NameIndex.lowerBound("a"), // iterate over name starting with "a",
  direction: "prev" // descending order
})

// iterate over objects of index / objectstore
nameIndex.iterator.iterateValues(async (cursor)=>{
```

wouldn't one typically want to `await` `nameIndex.getAll`, `nameIndex.iterator.iterateKeys` and `nameIndex.iterator.iterateValues`?

**Comment by qualified-cactus** [comment]
Thanks for the feedback, I will update the doc.

**Comment by qualified-cactus** [comment]
fixed in commit 6bb7e8c73f865316c345cf8c56fe1a305b19dbf3
